### PR TITLE
Testfix & server-side batch trashing

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

**Comment by frankrousseau** [comment]
Thank you for this addition. Could you add comments?

File: server/controllers/messages.coffee [hunk 1]
```diff
@@ -363,6 +364,38 @@ module.exports.send = (req, res, next) ->
         res.send out
 
 
+module.exports.batchTrash = (req, res, next) ->
+    ids = req.body.ids
+    unless ids?.length and ids.length > 0
+        return next new BadRequest "no ids in request's body"
```
Comment (frankrousseau) [inline server/controllers/messages.coffee]: You should not put return in the middle of a function.

File: server/controllers/messages.coffee [hunk 2]
```diff
@@ -363,6 +364,38 @@ module.exports.send = (req, res, next) ->
...
+        # in background, proceed to delete
+        async.eachSeries ids, (id, cb) ->
+            Message.moveToTrash account, id, (err) ->
+                reporter.onError err if err
+                reporter.addProgress 1
+                cb null # loop anyway
+        , (err, messages) ->
+            reporter.onDone()
+            log.info "BATCH TRASH #{reporter.id} COMPLETE"
+
```
Comment (frankrousseau) [inline server/controllers/messages.coffee]: Only two blank lines are required between functions.

**Comment by frankrousseau** [comment]
Could you add comments to your code?

**Comment by aenario** [comment]
done

**Comment by frankrousseau** [comment]
Great, thank you!

**Pull request merged**